### PR TITLE
Adds admin credentials and trusted domains provisioning

### PR DESCRIPTION
**Pull request by ypaquette**

### CODE REVIEW

**Comment by ypaquette** [comment]
Adds admin credentials and trusted domains provisioning

**Comment by shoekstra** [comment]
Looks awesome!  But the tests fail.

Could you fix this please?  Looks like the template generation needs fixing.

File: README.markdown [hunk 1]
```diff
@@ -200,7 +200,15 @@ The `owncloud` class configures all possible options for this module. With defau
...
-Sets the directory user data will be stored in. It is not recommended to keep this in the default location (as a sub directory of the application document root) and it should be moved out of the document root before making your ownCloud instance accessible via the internet. Defaults to '/var/www/owncloud/data' on Debian based systems and `/var/www/html/owncloud/data` on RedHat based systems.
+Sets the directory user data will be stored in. It is not recommended to keep this in the default location (as a sub directory of the application document root) and it should be moved out of the document root before making your ownCloud instance accessible via the internet. Defaults to `/var/www/owncloud/data` on Debian based systems and `/var/www/html/owncloud/data` on RedHat based systems.
+
+##### `admin_user`
+
+Optionally set the admin user in the ownCloud configuration (using `admin_pass` as the password). If not set, OwnCloud will ask for admin credentials upon first connection.  
+
+##### `admin_pass`
+
+Optionally set the admin password in the ownCloud configuration.
```
Comment (shoekstra) [inline README.markdown]: Could you move these parameter descriptions above the `datadirectory` description to keep in alphabetical order please.

File: templates/autoconfig.php.erb [hunk 2]
```diff
@@ -7,4 +11,7 @@ $AUTOCONFIG = array(
   "dbhost"        => "<%= @db_host %>",
   "dbtableprefix" => "<%= @db_table_prefix %>",
   "directory"     => "<%= @datadirectory %>",
-);
+<% if !@trusted_domains.empty? -%>
+  "trusted_domains" => <%= @trusted_domains %>,
+<% end -%>
+);
```
Comment (shoekstra) [inline templates/autoconfig.php.erb]: This should be changed to `unless @trusted_domains.empty?`

File: templates/autoconfig.php.erb [hunk 3]
```diff
@@ -1,5 +1,9 @@
 <?php
 $AUTOCONFIG = array(
+<% if !@admin_user.empty? -%>
+  "adminlogin"    => "<%= @admin_user %>",
+  "adminpass"     => "<%= @admin_pass %>",
+<% end -%>
   "dbtype"        => "<%= @db_type %>",
```
Comment (shoekstra) [inline templates/autoconfig.php.erb]: This should be changed to `unless @admin_user.empty? and @admin_pass.empty?` to avoid an empty adminpass parameter.

**Comment by svigneux** [comment]
```
rspec ./spec/classes/owncloud_spec.rb:223 # owncloud supported operating systems on debian-8-x86_64 owncloud class without any parameters should create $datadirectory if it doesn't exist, create database, populate autoconfig.php.erb with default values, remove skeleton dirs
```

@shoekstra : Hi there, we are having issues fixing the tests. We think the code issue is related only to the file format required by this section of owncloud spec test:  

```
            default_autoconfig = <<-EOF.gsub(/^ {14}/, '')
              <?php
              $AUTOCONFIG = array(
                "dbtype"        => "mysql",
                "dbname"        => "owncloud",
                "dbuser"        => "owncloud",
                "dbpass"        => "owncloud",
                "dbhost"        => "localhost",
                "dbtableprefix" => "",
                "directory"     => "#{datadirectory}",
              );
            EOF
```
If we apply the catalog without before and after our changes; the file format is exactly the same. 

Perhaps the new <% unless %> conditions are not parsed the same way running in a rake test or compiled by puppet.  

Is there any ```bundle exec rake test ``` parameters we could use to get more detailed information about where the template fails? Thanks!

**Comment by shoekstra** [comment]
Hello @svigneux - the problem is the missing line ending on the last line; not sure what editor you are using but each file that has been modified no longer has a line ending on the last line.  When I fixed this in the ERB the tests pass.

I would've thought this would come up in the rspec diff like other changes would, but I guess it's because the content looks the same.  When I fix it and use `git diff` I get:

```
$ git diff
diff --git a/templates/autoconfig.php.erb b/templates/autoconfig.php.erb
index 83a6171..ab60746 100644
--- a/templates/autoconfig.php.erb
+++ b/templates/autoconfig.php.erb
@@ -16,4 +16,4 @@ $AUTOCONFIG = array(
 <% unless @trusted_domains.empty? -%>
   "trusted_domains" => <%= @trusted_domains %>,
 <% end -%>
-);
\ No newline at end of file
+);
```

Can you also please add two new tests - one for `$admin_user` and `$admin_pass`, and one for `trusted_domains`.  You can use [this test](https://github.com/shoekstra/puppet-owncloud/blob/develop/spec/classes/owncloud_spec.rb#L314-L328) as an example.

Cheers!

**Comment by ypaquette** [comment]
I have some difficulties with our success for the trusted_domains test.
Is there anything you could suggest us in order to get it going?

**Comment by antoinedeschenes** [comment]
@ypaquette Small syntax changes in pull request allows to pass Travis build. Might be worth squashing.

**Comment by shoekstra** [comment]
Squashing would be ideal, but otherwise all looks good so ready to merge when you are.

**Comment by ypaquette** [comment]
Commits squashed, tests successfull

**Comment by shoekstra** [comment]
Awesome, nice work!

**Pull request merged**